scripts/aks-with-registry.sh

```diff
@@ -0,0 +1,202 @@
+    --location "${AZURE_LOCATION}" \
+    --output none --only-show-errors \
+    --tags creationTimestamp="${TIMESTAMP}" jobName="${JOB_NAME}" buildProvenance="${BUILD_PROVENANCE}"
+  fi
+
+  aks_exists=$(az aks show --name "${MGMT_CLUSTER_NAME}" --resource-group "${AKS_RESOURCE_GROUP}" 2>&1 || true) # true because we want to continue if the command fails
+  if echo "$aks_exists" | grep -E -q "Resource(NotFound|GroupNotFound)"; then
+    echo "creating aks cluster ${MGMT_CLUSTER_NAME} in the resource group ${AKS_RESOURCE_GROUP}"
+    az aks create --name "${MGMT_CLUSTER_NAME}" \
+    --resource-group "${AKS_RESOURCE_GROUP}" \
+    --location "${AZURE_LOCATION}" \
+    --kubernetes-version "${KUBERNETES_VERSION}" \
+    --node-count "${AKS_NODE_COUNT}" \
+    --node-vm-size "${AKS_NODE_VM_SIZE}" \
+    --node-resource-group "${AKS_NODE_RESOURCE_GROUP}" \
+    --vm-set-type VirtualMachineScaleSets \
+    --generate-ssh-keys \
+    --network-plugin azure \
+    --tags creationTimestamp="${TIMESTAMP}" jobName="${JOB_NAME}" buildProvenance="${BUILD_PROVENANCE}" \
+    --output none --only-show-errors;
+  elif echo "$aks_exists" | grep -q "${MGMT_CLUSTER_NAME}"; then
+    echo "cluster ${MGMT_CLUSTER_NAME} already exists in RG ${AKS_RESOURCE_GROUP}, moving on"
+  else
+    echo "error : ${aks_exists}"
+    exit 1
+  fi
+
+  # check and save kubeconfig
+  echo "saving credentials of cluster ${MGMT_CLUSTER_NAME} in ${REPO_ROOT}/${MGMT_CLUSTER_KUBECONFIG}"
+  az aks get-credentials --name "${MGMT_CLUSTER_NAME}" --resource-group "${AKS_RESOURCE_GROUP}" \
+  --file "${REPO_ROOT}/${MGMT_CLUSTER_KUBECONFIG}" --only-show-errors
+
+  az aks get-credentials --name "${MGMT_CLUSTER_NAME}" --resource-group "${AKS_RESOURCE_GROUP}" \
+  --overwrite-existing --only-show-errors
+
+  # echo "fetching Client ID for ${MGMT_CLUSTER_NAME}"
+  AKS_MI_CLIENT_ID=$(az aks show -n "${MGMT_CLUSTER_NAME}" -g "${AKS_RESOURCE_GROUP}" --output json \
+  --only-show-errors | jq -r '.identityProfile.kubeletidentity.clientId')
+  export AKS_MI_CLIENT_ID
+  echo "mgmt client identity: ${AKS_MI_CLIENT_ID}"
+  echo "${AKS_MI_CLIENT_ID}" > "${AZURE_IDENTITY_ID_FILEPATH}"
+
+  # echo "fetching Object ID for ${MGMT_CLUSTER_NAME}"
+  AKS_MI_OBJECT_ID=$(az aks show -n "${MGMT_CLUSTER_NAME}" -g "${AKS_RESOURCE_GROUP}" --output json \
+  --only-show-errors | jq -r '.identityProfile.kubeletidentity.objectId')
+  export AKS_MI_OBJECT_ID
+  echo "mgmt object identity: ${AKS_MI_OBJECT_ID}"
+
+  # echo "fetching Resource ID for ${MGMT_CLUSTER_NAME}"
+  AKS_MI_RESOURCE_ID=$(az aks show -n "${MGMT_CLUSTER_NAME}" -g "${AKS_RESOURCE_GROUP}" --output json \
+  --only-show-errors | jq -r '.identityProfile.kubeletidentity.resourceId')
+  export AKS_MI_RESOURCE_ID
+  echo "mgmt resource identity: ${AKS_MI_RESOURCE_ID}"
+
+  # save resource identity name and resource group
+  MANAGED_IDENTITY_NAME=$(az identity show --ids "${AKS_MI_RESOURCE_ID}" | jq -r '.name')
+  # export MANAGED_IDENTITY_NAME
```

Comment:
Are these `export` and `echo` lines that are commented out intended to be here, or is it just for debugging?

Reply:
Good spot, I added them for debugging, we dont need them anymore. Removing.